components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/util/OAuth2Util.java

```diff
@@ -2974,4 +2981,77 @@ public static void addTokenDOtoCache(AccessTokenDO newTokenBean) throws Identity
             }
         }
     }
+
+    /**
+     * Used to get the authenticated IDP name from a user.
+     *
+     * @param user Authenticated User.
+     * @return Returns the authenticated IDP name.
+     */
+    public static String getAuthenticatedIDP(AuthenticatedUser user) {
+
+        String authenticatedIDP;
+        if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled()) {
+            if (!OAuthServerConfiguration.getInstance().isMapFederatedUsersToLocal() && user.isFederatedUser()) {
+                authenticatedIDP = user.getFederatedIdPName();
+                if (log.isDebugEnabled()) {
+                    log.debug("IDP_ID column is available. User is federated and not mapped to local users. " +
+                            "Authenticated IDP is set to:" + authenticatedIDP);
+                }
+            } else {
+                authenticatedIDP = FrameworkConstants.LOCAL_IDP_NAME;
+                if (log.isDebugEnabled()) {
+                    log.debug("IDP_ID column is available. Authenticated IDP is set to:" + authenticatedIDP);
+                }
+            }
+        } else {
+            authenticatedIDP = user.getFederatedIdPName();
+            if (log.isDebugEnabled()) {
+                log.debug("IDP_ID column is not available. Authenticated IDP is set to:" + authenticatedIDP);
+            }
+        }
+
+        return authenticatedIDP;
+    }
+
+    /**
+     * Used to get the user store domain name from a user.
+     *
+     * @param user Authenticated User.
+     * @return Returns the sanitized user store domain.
+     */
+    public static String getUserStoreDomain(AuthenticatedUser user) {
+
+        String userDomain;
+        if (OAuth2ServiceComponentHolder.isIDPIdColumnEnabled() &&
+                !OAuthServerConfiguration.getInstance().isMapFederatedUsersToLocal() && user.isFederatedUser()) {
+            if (log.isDebugEnabled()) {
+                log.debug("IDP_ID column is available. User is federated and not mapped to local users.");
```

Comment:
shall we log some context information as well? e.g. username and tenant domain

Reply:
Improved the logs. Added user.toString() which will contain the fully-qualified username.